### PR TITLE
[bitnami/etcd]: Automatic etcd defragmentation

### LINKED ISSUE #29619: [bitnami/etcd] Add etcd-defrag CronJob for Etcd Maintenance
### Name and Version

bitnami/etcd

### What is the problem this feature will solve?

Over time, as DB keys are added, updated, and deleted, the etcd database becomes **fragmented**. This fragmentation can result in inefficient use of disk space, with consequent degraded performance. In our case scenario (etcd deployed with apisix) this eventually caused the etcd nodes to crash.

According to the [v3.5 docs](https://etcd.io/docs/v3.5/op-guide/maintenance/), etcd needs **periodic maintenance**.

### What is the feature you are proposing to solve the problem?

To solve the fragmentation issue, etcd provides the `etcdctl defrag` command, which blocks the system from reading and writing data if the etcd member is running.

**Feature request**: add a chart template to include a **CronJob** which performs **automatic periodic etcd defragmentation**, possibly configurable via parameters.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by mikyll**
### Description of the change

The change adds the `defrag` section, providing the possibility to enable the automatic defragmentation via the `etcdctl defrag` utility.
This creates a CronJob that periodically run `etcdctl defrag` on all the available endpoints. If this default behaviour is not suitable for the wanted scenario, a custom command can be set through `defrag.cronjob.command`.

### Benefits

Etcd maintenance (key history compaction and storage defragmentation) feasible directly from the chart configuration > lowers the resources used.

### Possible drawbacks

None that comes to my mind, since it's not enabled by default.

### Applicable issues

- fixes #29619 

### Additional information

The defragmentation is especially useful if combined with automatic compaction of key history. Etcd keeps a history of keys revisions, which takes up some space even if the keys are deleted. Setting the auto compaction allows to clear the disk space. However, after this operation, the storage can be fragmented, consuming space even if it's free to use. The process of defragmentation releases this storage space back to the file system.

Further information on etcd defragmentation can be found [here](https://etcd.io/docs/v3.5/op-guide/maintenance/#defragmentation).

#### Some Context

We encountered this issue while running etcd instances as part of our apisix deployment on Google Kubernetes Engine. Since the chart didn't provide any information about [etcd maintenance](https://etcd.io/docs/v3.5/op-guide/maintenance/), we initially didn't even bother. That was until our nodes crashed due to insufficient disk space. After further investigation, we discovered that the problem was linked to how etcd handles keys history and that the automatic compaction isn't enough. Since the chart didn't provide support for automatic defragmentation, we patched the issue via a "static" template CronJob.
However, we think that this is a quite important feature, and deserves to be included in the chart 🙂

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by dgomezleon — APPROVED** [review]
LGMT. Thanks a lot for your contribution

**Comment by mikyll** [comment]
@dgomezleon I realized I forgot to add a **check** to verify whether key history **autocompaction is enabled** before activating the defragmentation CronJob. During testing, we found that if defragmentation is enabled without key history autocompaction, the CronJob occasionally fails.

Possible solutions:

- allow `defrag.enabled == true` only if autocompaction is enabled;
- force autocompaction when `defrag.enabled == true`;
- only add comment/documentation snippet in defrag/autocompaction sections;

**Comment by dgomezleon** [comment]
> Since to me this doesn't seem a big issue - compared to the utility of that feature - and the defrag CronJob is widely configurable, I would just:
> 
> 1. add a note in the `defrag` values description;
> 2. (optionally) show a info/warning message if it's enabled without autoCompaction (similar to what happens for `resourcesPreset`).
> 
> Sounds good? 🙂

I agree.  Adding a note with this info should be enough. Thanks

**Comment by mikyll** [comment]
@dgomezleon I followed DCO instructions (my git client decided to remove the automatic signoff 😅) but I think I messed up something, I ended up signing other commits... Maybe it's better if I remake a clean PR?

File: bitnami/etcd/Chart.yaml [hunk 1]
```diff
@@ -32,4 +32,4 @@ maintainers:
 name: etcd
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/etcd
-version: 10.3.1
+version: 10.4.1
```
Comment (dgomezleon) [inline bitnami/etcd/Chart.yaml:35]:
```suggestion
version: 10.4.0
```

**Comment by dgomezleon** [comment]
> @dgomezleon I followed DCO instructions (my git client decided to remove the automatic signoff 😅) but I think I messed up something, I ended up appending my signoff to other commits... Maybe it's better if I remake a clean PR?

Yes no problem. You can create a separate PR if you prefer :)

**Pull request closed without merging**